redis.go

```diff
@@ -280,34 +280,10 @@ func (c *baseClient) withConn(
 	if err != nil {
 		return err
 	}
-
 	defer func() {
 		c.releaseConn(ctx, cn, err)
 	}()
-
-	done := ctx.Done() //nolint:ifshort
-
-	if done == nil {
-		err = fn(ctx, cn)
-		return err
-	}
-
-	errc := make(chan error, 1)
-	go func() { errc <- fn(ctx, cn) }()
-
-	select {
-	case <-done:
-		_ = cn.Close()
-		// Wait for the goroutine to finish and send something.
-		<-errc
-
-		err = ctx.Err()
-	case err = <-errc:
-	}
-	if err != nil {
-		return fmt.Errorf("execute with conn: %w", err)
-	}
-	return nil
+	return fn(ctx, cn)
```

Comment:
is this returned error being logged by the caller? if not, it's a good idea to add log line here.

Reply:
not going to add log lines to the library. The users of the library are responsible for logging its errors (e.g. when we use in the `ts-api`)